pkg/syncer/apis/edge/v1alpha1/edge-syncconfig.go

```diff
@@ -0,0 +1,78 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +crd
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:resource:scope=Cluster,shortName=esc
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type EdgeSyncConfig struct {
+	/*
+	 */
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   EdgeSyncConfigSpec   `json:"spec,omitempty"`
+	Status EdgeSyncConfigStatus `json:"status,omitempty"`
+}
+
+// EdgeSyncConfigSpec defines the desired state of EdgeSyncConfig
+type EdgeSyncConfigSpec struct {
+	// Downsncyed resource list
+	DownSyncedResources []EdgeSyncConfigResource `json:"downSyncedResources,omitempty"`
+
+	// Upsncyed resource list
+	UpSyncedResources []EdgeSyncConfigResource `json:"upSyncedResources,omitempty"`
+
+	// Conversions
+	Conversions []EdgeSynConversion `json:"conversions,omitempty"`
+}
+
+// Resource specifies down/up synced resource with exact GVK and name (and namespace if not cluster scoped resource)
+type EdgeSyncConfigResource struct {
```

Comment:
FYI this is not a typical usage of the word "Resource" in the world of Kubernetes code.  While in the wide world of the web, "resource" is anything that has a URL or URI, in Kubernetes code the word "resource" usually has a more specific meaning: a kind of object.  That is, a collection of objects rather than a single object.  Of course, the code base is not entirely consistent in this regard.  The public-facing documentation is worse, it is definitely mixed-up in this regard.

I recommend using the term "object" in the code when an individual API object is meant, reserving "resource" for a kind of object.

Here is one place where the rubber meets the road.  As you know, "kind" and "resource" are two equivalent but distinct way of identifying a sort of object.  In the XXXResource struct at hand, one would expect that the Kind and Name fields hold the kind and resource, respectively.  For example, as in https://github.com/kubernetes/apimachinery/blob/v0.26.3/pkg/apis/meta/v1/types.go#L1081 .  But that is not what is going on here.